packages/desktop-client/src/components/accounts/Account.js

```diff
@@ -105,6 +112,33 @@ function AllTransactions({ account = {}, transactions, filtered, children }) {
     }));
   }, [schedules, accountId]);
 
+  let runningBalance = useMemo(() => {
+    if (!showBalances) {
+      return 0;
+    }
+
+    // balances is in descending order so latest order is at the start of the array.
+    return balances ? Object.values(balances)[0]?.balance ?? 0 : 0;
```

Comment:
Could this be something more like `balances[transactions[0].id].balance`? Relying on the order of the object like this feels wrong to me.

Reply:
This makes much more sense. Updated.